capstone-frontend/src/components/CreateList.jsx

```diff
@@ -1,41 +1,103 @@
 import React, { Component } from 'react';
 import { withRouter } from 'react-router-dom'
-import { Button, Form, Spinner, Modal } from 'react-bootstrap'
+import { Button, Form, Spinner, Modal, Col, Row } from 'react-bootstrap'
 
 class CreateList extends Component {
 
   constructor(props) {
     super(props)
 
     this.state = {
-      loading: false,
-      showModal: false
+      creatingBookList: false, // For Spinner
+      fetchingBooks: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: "",
+      searchResults: [],
+      displayBooks: false,
+      addedBooksIDs: [],
+      addedBooks: []
     }
   }
 
-  handleSubmit = async () => {
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingBooks: true })
+
+    let searchResults = await fetch(`https://8080-bbaec244-5a54-4467-aed6-91c386e88c1a.ws-us02.gitpod.io/api/search?searchTerm=${searchTerm}&maxResults=${4}`)
+      .then(response => response.json())
+      .catch(err => console.log(err));
+
+    if (typeof searchResults === "undefined"){
+      searchResults = [];
+    }
+
+    this.setState({ searchResults, displayBooks: true, fetchingBooks: false })
+  }
+
+  handleSearchTermChange = (event) => {
+
+    if (this.state.typingTimeout) {
+      clearTimeout(this.state.typingTimeout);
+    }
+
+    this.setState({
+      searchTerm: event.target.value,
+      typingTimeout: setTimeout(async () => {
+        await this.getBooks(this.state.searchTerm)
+      }, 500)
+    })
+  }
+
+  addBookToList = (book) => {
+    this.state.addedBooksIDs.push(book.id);
+    this.state.addedBooks.push(book);
+
+    // Rerender
+    this.setState({ addedBooksIDs: this.state.addedBooksIDs, addedBooks: this.state.addedBooks })
+  }
 
-    this.setState({ loading: true })
+  removeBookFromList = (book) => {
 
-    const name = document.getElementById("createBookListForm").value
+    const indexId = this.state.addedBooksIDs.indexOf(book.id);
+    this.state.addedBooksIDs.splice(indexId, 1);
+
+    const indexBook = this.state.addedBooks.indexOf(book);
+    this.state.addedBooks.splice(indexBook, 1)
+
+    // Rerender
+    this.setState({ addedBooksIDs: this.state.addedBooksIDs, addedBooks: this.state.addedBooks })
+  }
+
+  handleSubmit = async () => {
+
+    let name = document.getElementById("form-booklist-name").value
     const userID = window.localStorage.getItem("userID")
 
+    if (name === "") {
+      alert("No name provided. Using default: 'To Read'")
+      name = "To Read"
+    }
+
+    this.setState({ creatingBookList: true })
+
     const newBooklist = {
       "userID": userID,
-      "name": name
+      "name": name,
+      "gbookIDs": this.state.addedBooksIDs ? this.state.addedBooksIDs : []
```

Comment:
Wouldn't `"gbookIDs": this.state.addedBooksIDs` have the same functionality as this ternary? If that is empty then it would be the empty array you use in your "else" case anyways.